etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -230,15 +313,87 @@ ExecStop=@sbindir@/zfs unload-key '${dataset}'"   > "${dest_norm}/${keyloadunit}
       "${dataset}" >/dev/kmsg
   fi
 
-  # If the mountpoint has already been created, give it precedence.
+  if [ -n "${p_systemd_wantedby}" ] && \
+      [ "${p_systemd_wantedby}" != "-" ] ; then
+    p_systemd_noauto="on"
+    if [ "${p_systemd_wantedby}" = "none" ] ; then
+      wantedby=""
+    else
+      wantedby="${p_systemd_wantedby}"
+      before="${before} ${wantedby}"
+    fi
+  fi
+
+  if [ -n "${p_systemd_requiredby}" ] && \
+      [ "${p_systemd_requiredby}" != "-" ] ; then
+    if [ "${p_systemd_requiredby}" = "none" ] ; then
+      requiredby=""
+    else
+      p_systemd_noauto="on"
```

Comment:
See comment above. Also: please move the `noauto` change so that it also applies if `requiredby=none` (as you have for `wantedby`).  This will make it match the behavior of `systemd-fstab-generator`.

Reply:
@InsanePrawn Maybe this comment got skipped over. Why aren't we setting `p_systemd_noauto="on"` also when `required-by` is `none`?

Reply:
Fixed, the behavior is now the same for `wanted-by` and `require-by`